soroban-sdk/src/env.rs

```diff
@@ -373,23 +378,19 @@ impl Env {
 
         let rf = Rc::new(EmptySnapshotSource());
         let storage = internal::storage::Storage::with_recording_footprint(rf);
-        let env_impl = internal::EnvImpl::with_storage_and_budget(
-            storage,
-            internal::budget::Budget::default(),
-        );
-
+        let budget = internal::budget::Budget::default();
+        let env_impl = internal::EnvImpl::with_storage_and_budget(storage, budget.clone());
+        env_impl.switch_to_recording_auth();
```

Comment:
It seems a bit strange to me that we'd modify the env in this way, rather than use the pattern we have used for storage where we provide an implementation, and there is a recording or enforcing implementation. Is there a reason we can't use that existing pattern here?